docs/resource_sets.rst

```diff
@@ -91,3 +93,168 @@ Limitations
+not intersect on owned nodes, see the `Ownership`_ subsection.
+
+Service instance uniqueness
+***************************
+With full compiles, indexes serve as the identity of a service instance. The compiler then validates that no conflicting
+service instances can exist. With partial compiles this validation is lost because only one service instance will be present
+in the model. However, it is still crucial that such conflicts do not exist. Put simply, we need to make sure that a partial
+compile succeeds only if a full compile would succeed as well. This subsection deals solely with the uniqueness of service
+instances. The `Ownership`_ subsection then deals with safe refinements into resources.
+
+To ensure service instance definitions are distinct, the model must make sure to do appropriate validation on the full set of
+definitions. When doing a partial compile, the model must verify that the service instance it is compiling for, has a different
+identity from any of the previously defined service instances. This can be achieved by externally checking against some sort of
+inventory that there are no matches for any set of input attributes that identify the instance.
+
+The current implementation of partial compiles does not provide any helpers for this verification. It is the responsibility of
+the model developer or the tool/extension that does the export to ensure that no two service instances can be created that are
+considered to have the same identity by the model.
+
+Ownership
+*********
+A resource can safely be considered owned by a service instance if it could never be created by another service instance. There
+are two main mechanisms that can be used to provide this guarantee. One is the use of indexes on appropriate locations, the
+other is the use of some external distributor of unique values (e.g. a plugin to generate a UUID or to allocate values in an
+inventory).
+
+In either case, the goal is to make sure that any object that is marked as owned by a service instance, is unique to that
+instance. In the index case we do so by making sure the object's identity is in fact completely derived from the identity of
+the service instance. In the case where unique values are externally produced/allocated, responsibility for uniqueness falls
+to the plugin that produces the values.
+
+Generally, for every index on a set of attributes of an owned resource, at least one of the fields must be either derived from
+the identity of the service instance, or allocated in a safe manner by a plugin as described above. The same goes for every
+pair of resource id and agent. If the former constraint is not met, a full compile might fail, while if the latter is not met,
+the export will be rejected because two services are trying to configure the same resources.
+
+For example, consider the example model from before. If two networks with two routers each would be created, they would result
```

Comment:
This confuses me, because routers tend to connect networks, they can never be contained in them. (interfaces of routers can be in networks)

Reply:
I'll modify the example to use `Host` rather than `Router`.